parisbs/play-store-api
# Feature/environment setup

#### What does this PR do?
* Add base environment setup
* Add base unit tests
* Add coverage and linter

#### Where should the reviewer start?
1. requirements.txt
2. config.py
3. play_store_api/
4. tests/
5. Makefile
6. Other files

#### How should this be manually tested?
1. `cp .env.example .env` and modify variables
2. `make install`
3. Create virtual environment using `python -m venv [virtual_environment_path]`
4. `source [virtual_environment_path]/bin/activate`
5. `make test-all`
6. `make run`
7. Go to `http://localhost:[API_PORT]/v1/health`
8. Message 'OK' should be displayed

#### Any background context you want to provide?
* Requires python 3.X+
* Default port is 80, you can change this on .env file
* Make install add pre-commit hook to launch tests, coverage and linter

#### SW Checklist


#### QA Checklist

- [x] Project builds without errors locally.
- [ ] Two peer code reviews were performed.
- [ ] Feature/Fix tested against UX designs.
- [ ] Feature/Fix tested against user stories which contain Happy Path and Alternate Paths.
- [ ] Feature passes internal UAT by PjM or TL.
- [x] Relevant documentation updated: JSDocs / Confluence.
